### PR TITLE
Sample react app deployed in jaggeryapps + server-side routing

### PR DESCRIPTION
**Pull request by kavishkafernando**
## Purpose
This PR includes a Sample react app deployed in jaggeryapps (Publisher).
This PR also includes server-side routing for the react-app.
All the files included in the react app folders are sample files and should be replaced.
jaggery.conf and public/jaggery/index.jag should not be replaced.
org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new folder contains all the react resources.

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/index.jsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (npamudika) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/index.jsx]: Shall we update the licence year? And lets include licence headers for the other files too.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (npamudika) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/webpack.config.js]: Same here

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/pom.xml [hunk 3]
```diff
@@ -167,7 +167,102 @@
...
+                                    <name>org.wso2.carbon.p2.category.type</name>
+                                    <value>server</value>
+                                </advice>
+                                <advice>
+                                    <name>org.eclipse.equinox.p2.type.group</name>
+                                    <value>server</value>
+                                </advice>
+                            </adviceFileContents>
+                            <bundles>
+
```
Comment (npamudika) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/pom.xml]: Shall we remove this extra line?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/pom.xml [hunk 4]
```diff
@@ -167,7 +167,102 @@
...
+                                <argument>${npm.build.command}</argument>
+                            </arguments>
+                        </configuration>
+                    </execution>
+                </executions>
+                <configuration>
+                    <workingDirectory>${npm.working.dir}</workingDirectory>
+                </configuration>
+            </plugin>
+
```
Comment (npamudika) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/pom.xml]: This too

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/public/js/hello.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+<body>
+
+<h2>JavaScript in Head</h2>
+
+<p id="demo">A Paragraph.</p>
+
+<button type="button" onclick="myFunction()">Try it</button>
+
+</body>
+</html>
```
Comment (TanyaM) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/public/js/hello.js]: New line required

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/index.jsx (hunk 1 above)
Comment (TanyaM) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/index.jsx]: Wrong license year

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/webpack.config.js (hunk 2 above)
Comment (TanyaM) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/webpack.config.js]: Wrong license year

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/index.jsx (hunk 1 above)
Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/index.jsx]: fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/webpack.config.js (hunk 2 above)
Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/webpack.config.js]: fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/pom.xml (hunk 3 above)
Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/pom.xml]: removed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/pom.xml (hunk 4 above)
Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/pom.xml]: removed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/public/js/hello.js (hunk 5 above)
Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/public/js/hello.js]: fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/index.jsx (hunk 1 above)
Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/index.jsx]: fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/webpack.config.js (hunk 2 above)
Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/webpack.config.js]: fixed

**Pull request merged**